### PR TITLE
Made input[type="color"] squared instead of blocking across

### LINKED ISSUE #18786: <input type="color" class="form-control"> as colorpicker is displayed badly
When browsers such as Chrome and Firefox present a proper colorpicker UI for `<input type="color">`, if you use `.form-control` on the input, it's displayed kinda weirdly.

In Chrome, the padding scrunches the actual color swatch and makes it invisible:
chrome

In Firefox, the height of the input doesn't match the height of `<input type="text" class="form-control">`:
firefox

Safari doesn't support color inputs, so it shows a plain text input, which looks fine.
I haven't tested on IE/Edge.
CC: @mdo

### PR DESCRIPTION
**Pull request by jeradrutnam**
Fixes #18786.

Please refer the new PR: https://github.com/twbs/bootstrap/pull/21155

### CODE REVIEW

**Comment by cvrebert** [comment]
Given that [not all of our target browsers support `type=color`](http://caniuse.com/#feat=input-color), and the unsupported browsers use a text field instead, I'm doubtful that the resulting small square text field in those browsers would be usable.

**Comment by patrickhlauke** [comment]
wondering though if that's not more of a problem for the site/app author to worry about (i.e. i doubt that they'd use color input and expect users in non-supporting browsers to type in hex values manually, so they'd do feature detection at their end). this could be tackled with a note somewhere in the docs and the css?

**Comment by patrickhlauke** [comment]
something like "Note that browsers that do not support `<input type="color">` will revert to using a simple text entry input - however,  these browsers will still apply the styles for `input[type="color"]`, resulting in potentially unusable small text inputs..."

**Comment by jeradrutnam** [comment]
Hi,

I too believe developers should be aware of what html elements they use, when they do cross-browser development. So breaking the default behavior in supported browsers will end up developers writing overwrite class to fix it. 

So IMO its better we handle it for supported browser and then we can find a work around for others while handling with warning message in Docs.

Cheers!

**Comment by wolfy1339** [comment]
I would suggest adding it in the custom forms instead, also I have noticed that you included changes in the dist folder. Please exclude those changes

**Comment by cvrebert** [comment]
I agree it doesn't seem feasible to deal with browsers that don't support `<input type="color">`.
But since it's such a different widget (since it's non-textual), I'd argue for using a completely separate class instead of crowbaring the styles into `.form-control`.

**Comment by jeradrutnam** [comment]
Hi,

That make sense. I'll change it to a custom class name instead `.custom-color`. And exclude the dist folder as mention and update the PR. And update it in the docs as well ?

**Comment by cvrebert** [comment]
@jeradrutnam Sounds great!

**Comment by jeradrutnam** [comment]
I have updated the PR. Please review and merge.

Docs preview,
forms bootstrap

File: docs/components/forms.md [hunk 1]
```diff
@@ -889,6 +883,20 @@ Custom `<select>` menus need only a custom class, `.custom-select` to trigger th
 
 Custom selects degrade nicely in IE9, receiving only a handful of overrides to remove the custom `background-image`. **Multiple selects (e.g., `<select multiple>`) are not currently supported.**
 
+### Color picker
+
+{% callout danger %}
+#### Internet Explorer compatibility
```
Comment (cvrebert) [inline docs/components/forms.md]: Change this to "Browser compatibility"

File: docs/components/forms.md [hunk 2]
```diff
@@ -889,6 +883,20 @@ Custom `<select>` menus need only a custom class, `.custom-select` to trigger th
...
+
+While Bootstrap supports styling `<input type="color">` elements, Internet Explorer Edge 13 and below don’t support. Use custom JavaScript to handle it in these browsers.
```
Comment (cvrebert) [inline docs/components/forms.md]: Per Can I Use, the full list of relevant non-supporting browsers is:
IE, Edge 13, OS X Safari 9.1, iOS 9.3 Safari

File: docs/components/forms.md [hunk 3]
```diff
@@ -889,6 +883,20 @@ Custom `<select>` menus need only a custom class, `.custom-select` to trigger th
...
+{% endcallout %}
```
Comment (cvrebert) [inline docs/components/forms.md]: Add something like: "For details, see [Can I Use...](http://caniuse.com/#feat=input-color)"

**Comment by cvrebert** [comment]
Nice work. I don't think this is drastic enough to need to be deemed a custom form control, but I don't know what a good name for the class would be.

**Comment by cvrebert** [comment]
`color` also needs to be removed from the table in https://github.com/twbs/bootstrap/blob/v4-dev/docs/components/forms.md#form-controls

**Comment by cvrebert** [comment]
CC: @mdo for review

File: docs/components/forms.md (hunk 1 above)
Comment (jeradrutnam) [inline docs/components/forms.md]: Done, updated the PR. :)

**Comment by jeradrutnam** [comment]
Added new class '.form-control-color' to match '.form-control-file'. So this will avoid conflicting with the class '.custom-color' :)

File: docs/components/forms.md (hunk 3 above)
Comment (jeradrutnam) [inline docs/components/forms.md]: I have updated docs warning message as "For details, see [Can I Use...](http://caniuse.com/#feat=input-color)" as suggested.

**Comment by wolfy1339** [comment]
@jeradrutnam Is there a reason you closed this PR, it's in the Alpha 6 milestone now

**Comment by jeradrutnam** [comment]
@wolfy1339 Actually I didn't closed it. I just took a pull from the latest and fixed the conflict. And the PR has got closed automatically. And I can see the changes listed in my fork. https://github.com/twbs/bootstrap/compare/v4-dev...jeradrutnam:v4-dev?expand=1

But cannot see the file changes in this PR.

**Comment by Johann-S** [comment]
I think you accidentally revert you work, because they aren't any files in your PR :thinking:

**Comment by wolfy1339** [comment]
I made a new PR for you, #21155 
@jeradrutnam Just a tip, don't use any branches present on twbs/bootstrap, use your own names (can be anything, ex patch-1) to avoid any conflicts.
Another tip, use `git pull upstream/v4-dev --rebase` to pull in the latest changes from v4-dev without making a merge commit

**Comment by jeradrutnam** [comment]
@wolfy1339 Thanks a lot. Yes, I just realized that. It's my mistake. I will work on different brunch in the future. This happened because I did it sometime back and I couldn't keep the this brunch up to date.

**Pull request closed without merging**